ludwig/data/dataset_synthesizer.py

```diff
@@ -339,26 +382,24 @@ def generate_image(feature, save_as_numpy=False):
     image_filename = uuid.uuid4().hex[:10].upper() + ".png"
 
     # Save the image to disk either in a specified location/new folder
+    if not os.path.exists(destination_folder):
+        os.makedirs(destination_folder)
+    image_dest_path = os.path.join(destination_folder, image_filename)
     try:
-        if not os.path.exists(destination_folder):
-            os.makedirs(destination_folder)
-
-        image_dest_path = os.path.join(destination_folder, image_filename)
         # save_image(torch.from_numpy(img.astype("uint8")), image_dest_path)
         if save_as_numpy:
             with open(image_dest_path, "wb") as f:
                 np.save(f, img.detach().cpu().numpy())
         else:
             write_png(img, image_dest_path)
-
     except OSError as e:
-        raise OSError("Unable to create a folder for images/save image to disk." "{}".format(e))
+        raise OSError(f"Unable to create a folder for images/save image to disk: {e}")
```

Comment:
nit: same as above, because we should probably report creating folder-related failures separately from saving